kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/ChecksumStatsSuite.scala

```diff
@@ -172,9 +173,40 @@ class ChecksumStatsSuite extends DeltaTableWriteSuiteBase {
       engine: Engine,
       dataActions: CloseableIterable[Row]): TransactionCommitResult = {
     val result = txn.commit(engine, dataActions)
-    result.getPostCommitHooks
-      .stream()
-      .forEach(hook => hook.threadSafeInvoke(engine))
+    val checksumHook = result.getPostCommitHooks.stream().filter(hook =>
+      hook.getType == getPostCommitHookType).findFirst()
+    // When result.getVersion is 0, there will only be CHECKSUM_SIMPLE.
+    assert(result.getVersion == 0 || checksumHook.isPresent)
+    checksumHook.ifPresent(_.threadSafeInvoke(engine))
     result
   }
+
+  protected def getPostCommitHookType: PostCommitHookType
+}
+
+class ChecksumSimpleStatsSuite extends ChecksumStatsSuiteBase {
+  override def getPostCommitHookType: PostCommitHookType = PostCommitHookType.CHECKSUM_SIMPLE
+}
+
+class ChecksumFullStatsSuite extends ChecksumStatsSuiteBase {
+  override def getPostCommitHookType: PostCommitHookType = PostCommitHookType.CHECKSUM_FULL
+
+  override def checkCrcCorrect(
+      engine: Engine,
+      tablePath: String,
+      version: Long,
+      expectedFileCount: Long,
+      expectedTableSize: Long,
+      expectedFileSizeHistogram: FileSizeHistogram): Unit = {
+    super.checkCrcCorrect(
+      engine,
+      tablePath,
+      version,
+      expectedFileCount,
+      expectedTableSize,
+      expectedFileSizeHistogram)
+    // Delete the checksum after the validation, so that the subsequence commit
+    // will generate CHECKSUM_FULL hook.
+    deleteChecksumFileForTable(tablePath.stripPrefix("file:"), Seq(version.toInt))
```

Comment:
again not sure, but I think maybe it makes more semantic sense to do this instead before we commit?